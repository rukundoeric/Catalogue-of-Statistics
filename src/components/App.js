import React from 'react';
import '../assets/css/style.css';

function App() {
  return (
    <div className="content">
      <h1>Hi here!</h1>
    </div>
  );
}

export default App;
