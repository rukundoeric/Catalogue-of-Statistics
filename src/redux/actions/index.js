import axios from 'axios';
import { randomMeal, categories } from './_api';

export const getRandomMeal = () => async dispatch => {
  const { data: { meals } } = await axios.get(randomMeal);
  dispatch({
    type: 'GET_RANDOM_MEAL',
    payload: meals[0],
  });
};

export const getCategories = () => async dispatch => {
  const { data: { meals: payload } } = await axios.get(categories);
  dispatch({
    type: 'GET_CATEGORIES',
    payload,
  });
};
