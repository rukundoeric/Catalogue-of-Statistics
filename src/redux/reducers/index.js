import { combineReducers } from 'redux';

const meals = () => [];

export default combineReducers({
  meals,
});
