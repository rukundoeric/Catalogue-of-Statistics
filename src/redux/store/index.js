import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from '../reducers';

const { NODE_ENV } = process.env;
const middleware = [thunk];
const initialState = {};

const enableDevTools = NODE_ENV === 'development'
  ? composeWithDevTools(applyMiddleware(...middleware))
  : applyMiddleware(...middleware);

const store = createStore(rootReducer, initialState, enableDevTools);

export default store;
