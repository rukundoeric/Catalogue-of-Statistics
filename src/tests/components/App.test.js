/* eslint-disable react/prop-types */
import '@testing-library/jest-dom';
import { render, cleanup } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import App from '../../components/App';
import rootReducer from '../../redux/reducers';

afterEach(cleanup);

const initialState = {
  meals: [],
};

const store = createStore(rootReducer, initialState);

const Wrapper = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

it('Should match snapshot', () => {
  const { asFragment } = render(<App />, { wrapper: Wrapper });
  expect(asFragment(<App />)).toMatchSnapshot();
});
