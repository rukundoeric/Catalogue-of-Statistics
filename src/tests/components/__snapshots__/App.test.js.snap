// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    <h1>
      Hi here!
    </h1>
  </div>
</DocumentFragment>
`;
